test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -288,6 +308,172 @@ public void testRoutingQueueReplicationWithScopedTxn() throws Exception {
+                            .setReplicationType(Queue.ReplicationType.SNAPSHOT_SYNC)
+                            .addDestinations(destinationId)
+                            .setOpaquePayload(ByteString.copyFromUtf8("opaquetxn"+i))
+                            .buildPartial();
+                    txn.logUpdateEnqueue(snapSyncQ.getStreamUUID(), snapshotMessage, Collections.singletonList(TableRegistry.getStreamIdForStreamTag(CORFU_SYSTEM_NAMESPACE,
+                            SNAPSHOT_SYNC_QUEUE_TAG_SENDER_PREFIX + destinationId + "_" + DEFAULT_ROUTING_QUEUE_CLIENT)), corfuStore);
+                    addMarker(tail, true, destinationId, corfuStore, txn);
+                } else {
+                    Queue.RoutingTableEntryMsg deltaMessage = Queue.RoutingTableEntryMsg.newBuilder()
+                            .setReplicationType(Queue.ReplicationType.LOG_ENTRY_SYNC)
+                            .addDestinations(destinationId)
+                            .setOpaquePayload(ByteString.copyFromUtf8("opaquetxn"+i))
+                            .buildPartial();
+                    txn.logUpdateEnqueue(logEntryQ.getStreamUUID(), deltaMessage,
+                            Collections.singletonList(TableRegistry.getStreamIdForStreamTag(CORFU_SYSTEM_NAMESPACE,
+                            LOG_ENTRY_SYNC_QUEUE_TAG_SENDER_PREFIX + destinationId + "_" + DEFAULT_ROUTING_QUEUE_CLIENT)), corfuStore);
+                }
+                txn.commit();
+            }
+        }
+
+        // add end marker to denote end of snapshot sync
+        try(TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            addMarker(tail, false, DefaultClusterConfig.getSinkClusterIds().get(0), corfuStore, txn);
+            txn.commit();
+        }
+
+
+        // open the recvQ on SINK
+        Table<Queue.CorfuGuidMsg, Queue.RoutingTableEntryMsg, Queue.CorfuQueueMetadataMsg> recvQ = sinkCorfuStores.get(0).openQueue(
+                CORFU_SYSTEM_NAMESPACE, LogReplicationUtils.REPLICATED_RECV_Q_PREFIX + DefaultClusterConfig.getSourceClusterIds().get(0) + "_" + DEFAULT_ROUTING_QUEUE_CLIENT,
+                Queue.RoutingTableEntryMsg.class, TableOptions.builder().schemaOptions(
+                        CorfuOptions.SchemaOptions.newBuilder().addStreamTag(REPLICATED_QUEUE_TAG).build()).build());
+
+        // wait until the recv side count == data written on source
+        while(recvQ.count() < (numLogEntryUpdates + 1)) {
+
+        }
+
+        // the "+1" is for the dummy record that is written at the end of snapshot sync
+        assertThat(recvQ.count()).isEqualTo(numLogEntryUpdates + 1);
+
+
+        AtomicBoolean snapshotSyncData = new AtomicBoolean(true);
+        AtomicInteger countOfSnapshotSyncData = new AtomicInteger(0);
+        AtomicInteger countOfDeltaSyncData = new AtomicInteger(0);
+
+        // validate that the order is maintained.
+        recvQ.entryList().forEach(entry -> {
```

Comment:
we are not performing any cp/trim here so the order cannot be verified, right?  We can either add a cp/trim or compare the sequence numbers in the queue records.

Reply:
I realized that we don't need to simulate CP/Trim to verify the order, we can directly use `entryList` to check the order.
The `entryList` returns the records by sorting on the keys. This operation is what will be used on CP/trim as well. 

The test fails without this change, and on debug mode, we can see the snapshot data and log-entry data being interleaved since the keys have the sequence number of the SOURCE..